config/validator/service/proposer_settings.go

```diff
@@ -84,6 +84,11 @@ type ProposerSettings struct {
 	DefaultConfig *ProposerOption
 }
 
+// ShouldBeSaved goes through checks to see if the value should be saveable.
+func (settings *ProposerSettings) ShouldBeSaved() bool {
```

Comment:
Could you add more description in the comments ? Its hard to know what exactly these nil checks are for to someone not familiar with the code